backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/interim/invoice/InterimInvoiceCandidateService.java

```diff
@@ -139,24 +140,28 @@ public ImmutableSet<InvoiceCandidateId> createInterimInvoiceCandidatesFor(@NonNu
 
 		final ImmutableSet.Builder<InvoiceCandidateId> invoiceCandidateSet = ImmutableSet.builder();
 
-		inOutLines.forEach(
-				inOutLine -> {
-					final StockQtyAndUOMQty stockDeliveredQty = inOutBL.getStockQtyAndQtyInUOM(inOutLine);
-
-					final NewInvoiceCandidate newInvoiceCandidate = newInvoiceCandidateTemplate
-							.qtyOrdered(stockDeliveredQty)
-							.qtyDelivered(stockDeliveredQty)
-							.build();
-
-					final InvoiceCandidateId invoiceCandidateId = invoiceCandidateRepository.save(manualCandidateService.createInvoiceCandidate(newInvoiceCandidate));
-					invoiceCandidateSet.add(invoiceCandidateId);
-					modularContractLogService.setICProcessed(
-							ModularContractLogQuery.builder()
-									.contractType(LogEntryContractType.INTERIM)
-									.referenceSet(TableRecordReferenceSet.of(TableRecordReference.of(I_M_InOutLine.Table_Name, inOutLine.getM_InOutLine_ID())))
-									.build(),
-							invoiceCandidateId);
-				});
+		final StockQtyAndUOMQty initialStockQtyAndUOM = StockQtyAndUOMQty.builder()
+				.productId(productIdToInvoice)
+				.stockQty(Quantitys.createZero(stockUOM))
+				.uomQty(Quantitys.createZero(stockUOM))
+				.build();
+
+		final StockQtyAndUOMQty totalStockQtyAndUOMQty = inOutLines.stream().map(inOutLine -> inOutBL.getStockQtyAndQtyInUOM(inOutLine))
+
+				.reduce(initialStockQtyAndUOM, StockQtyAndUOMQty::add);
```

Comment:
shouldn't we do nothing in case totalStockQtyAndUOMQty is zero?

Reply:
* From the beginning, the method does not get to create any invoice candidate if there are no inout lines.
* If by any change the inout lines have the qty 0, then we will end up with an invoice candidate and an invoice line with qty 0. The line will be displayed in webui.
* We are not yet sure if the line will be displayed in the report. Probably not, but Mark is not sure yet.